src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts

```diff
@@ -134,18 +136,27 @@ function drawHorizontalBarChartValues(
       const point = dataset.data[i];
 
       const yPosition = point.y;
-      let xPosition = value < 0 ? point.x + point.width / 2 : point.x - point.width / 2;
-      xPosition = Math.min(xPosition, xMax);
-      xPosition = Math.max(xPosition, xMin);
+      const textWidth = computeTextWidth(ctx, displayValue, { fontSize: 12 }, "px");
+      const barWidth = Math.abs(point.x - xZero);
+
+      const PADDING = 3;
+      let xPosition: number;
+      if (barWidth < textWidth) {
+        xPosition = value < 0 ? xZero - textWidth / 2 - PADDING : xZero + textWidth / 2 + PADDING;
+      } else {
+        xPosition = value < 0 ? point.x + point.width / 2 : point.x - point.width / 2;
+        xPosition = Math.min(xPosition, xMax);
+        xPosition = Math.max(xPosition, xMin);
+      }
 
       // Avoid overlapping texts with same Y
       if (!textsPositions[yPosition]) {
         textsPositions[yPosition] = [];
       }
-      const textWidth = computeTextWidth(ctx, displayValue, { fontSize: 12 }, "px");
       for (const otherPosition of textsPositions[yPosition]) {
         if (Math.abs(otherPosition - xPosition) < textWidth) {
-          xPosition = otherPosition + textWidth + 3;
+          xPosition =
```

Comment:
the same issue exists for vertical stacked bar charts. Maybe we could also fix it here ?